pkg/controller/sparkapplication/sparkui.go

```diff
@@ -94,16 +103,24 @@ func createSparkUIIngress(app *v1beta2.SparkApplication, service SparkService, i
 	return &SparkIngress{
 		ingressName: ingress.Name,
 		ingressURL:  ingressURL,
+		annotations: ingress.Annotations,
+		ingressTLS:  ingress.Spec.TLS,
 	}, nil
 }
 
 func createSparkUIService(
 	app *v1beta2.SparkApplication,
 	kubeClient clientset.Interface) (*SparkService, error) {
-	portStr := getUITargetPort(app)
+	portStr := getUIServicePort(app)
```

Comment:
Can `getUIServicePort` return a integer port instead?

Reply:
done + I did the same to getUITargetPort